utilities/realsense-test/src/Module.cpp

```diff
@@ -40,54 +46,112 @@ bool Module::configure(yarp::os::ResourceFinder& rf)
+    w_H_cam << 0,  0, 1, 0,
+              -1,  0, 0, 0,
+               0, -1, 0, 0,
+               0,  0, 0, 1;
     return true;
 }
 
 
 bool Module::updateModule()
 {
-    cv::Mat bgr;
-    if (rsDev->getColorImage("D435i", bgr))
+    if (m_showImages)
     {
-        cv::namedWindow(colorImgName);
-        cv::imshow(colorImgName, bgr);
-        cv::waitKey(1);
-    }
+        cv::Mat bgr;
+        if (rsDev->getColorImage("D435i", bgr))
+        {
+            cv::namedWindow(colorImgName);
+            cv::imshow(colorImgName, bgr);
+            cv::waitKey(1);
+        }
 
-    cv::Mat depth;
-    if (rsDev->getColorizedDepthImage("D435i", depth))
-    {
-        cv::namedWindow(depthImgName);
-        cv::imshow(depthImgName, depth);
-        cv::waitKey(1);
-    }
+        cv::Mat depth;
+        if (rsDev->getColorizedDepthImage("D435i", depth))
+        {
+            cv::namedWindow(depthImgName);
+            cv::imshow(depthImgName, depth);
+            cv::waitKey(1);
+        }
 
 
-    cv::Mat ir;
-    if (rsDev->getInfraredImage("D435i", ir))
-    {
-        cv::namedWindow(irImgName);
-        cv::imshow(irImgName, ir);
-        cv::waitKey(1);
+        cv::Mat ir;
+        if (rsDev->getInfraredImage("D435i", ir))
+        {
+            cv::namedWindow(irImgName);
+            cv::imshow(irImgName, ir);
+            cv::waitKey(1);
+        }
     }
 
-    if (rsDev->getPointCloud("D435i", pc))
+    if (rsDev->getPointCloud("D435i", pcRaw))
     {
         viewer->removePointCloud("srcCloud");
-        if (pc != nullptr)
+        if (pcRaw != nullptr)
         {
-            viewer->addPointCloud(pc, "srcCloud");
+
+            viewer->addPointCloud(pcRaw, "srcCloud");
             viewer->spinOnce();
+
+            // first downsample then remove outliers
+            pclProc.downsample(pcRaw, pcProcessed);
+            pclProc.removeOutliers(pcProcessed, pcProcessed);
+
+            std::vector<pcl::PointCloud<pcl::PointXYZRGB>::Ptr> clusters;
+            std::vector<pcl::PointIndices> clusterIndices;
+            if (!pclProc.extractClusters(pcProcessed, clusterIndices, clusters))
+            {
+                yInfo() << "Extraction failure." ;
+            }
+            yInfo() << "Nr of clusters: " << clusters.size();
+            for (size_t idx = 0; idx < previousClusterSize; idx++)
+            {
+                if (viewerProcessed->contains(std::to_string(idx)))
+                {
+                    viewerProcessed->removePointCloud(std::to_string(idx));
+                }
+            }
+
+            for (size_t idx = 0; idx < clusters.size(); idx++)
+            {
+                auto pcTrans = pcl::make_shared<pcl::PointCloud<pcl::PointXYZRGB>>();
+                pclProc.transform(clusters[idx], w_H_cam, pcTrans);
+                viewerProcessed->addPointCloud(pcTrans, std::to_string(idx));
+            }
+            previousClusterSize = clusters.size();
+
+            viewerProcessed->spinOnce();
```

Comment:
Do you think that some of this logic can be moved to an ``Advanceable``?

Reply:
I'd rather maintain some fluidity with this class. Since we are dealing with some computationally heavy data-type, putting these methods together in a single advance will make it rigid and maybe an overkill for some tasks. 

We can maybe roll out independent `Advanceable` classes that uses these methods depending on our future need?

Reply:
Ok for me!